### PR TITLE
add userspace actions

### PR DESCRIPTION
**Pull request by harshsin**
Reviewer: @rlane

### CODE REVIEW

File: modules/action/module/src/action.c [hunk 1]
```diff
@@ -56,6 +56,21 @@ action_controller(struct action_context *ctx, uint64_t userdata)
     nla_nest_end(ctx->msg, action_attr);
 }
 
+/* Send the packet back to an upcall thread with the given userdata
+   on the specified netlink socket */
+void
+action_userspace(struct action_context *ctx, void *userdata, int datalen,
+                 uint32_t netlink_port)
+{
+    assert(datalen <= sizeof(uint64_t));
```
Comment (rlane) [inline modules/action/module/src/action.c]: This assert is unnecessary. The userdata can be larger than 8 bytes.

File: modules/action/module/src/action.c [hunk 2]
```diff
@@ -104,6 +119,28 @@ action_sample_to_controller(struct action_context *ctx, uint64_t userdata, uint3
     nla_nest_end(ctx->msg, sample_attr);
 }
 
+void
+action_sample_to_userspace(struct action_context *ctx, void *userdata, int datalen,
+                           uint32_t netlink_port, uint32_t probability)
+{
+    assert(datalen <= sizeof(uint64_t));
```
Comment (rlane) [inline modules/action/module/src/action.c]: Likewise, remove this assertion.

**Comment by rlane** [comment]
The existing controller and sample actions should become wrappers around the more generic functions you added.

**Comment by harshsin** [comment]
Thanks for taking a look. Pushed another commit.

**Comment by rlane** [comment]
ACK

**Comment by bsn-abat** [comment]
ABAT: START_MERGE:  Log file at http://trusty1/abat/2015.02.26.0058-m.ivs.master/abat.log

**Comment by bsn-abat** [comment]
ABAT: ACCEPT:  Successfully merged

In case you want to see the build log, check out :
Log file at http://trusty1/abat/2015.02.26.0058-m.ivs.master/abat.log
/cc @rlane

**Pull request merged**